hnn_core/parallel_backends.py

```diff
@@ -344,11 +355,21 @@ def simulate(self, net):
         if proc.returncode != 0:
             raise RuntimeError("MPI simulation failed")
 
+        if not data_length == len(self.proc_data_bytes):
+            raise RuntimeError("Failed to receive all data from the child MPI"
+                               " process. Expecting %d bytes, got %d" %
+                               (data_length, len(self.proc_data_bytes)))
+
         if len(self.proc_data_bytes) == 0:
             raise RuntimeError("MPI simulation didn't return any data")
 
-        # decode base64 object
-        data_pickled = codecs.decode(self.proc_data_bytes, "base64")
+        # decode base64 byte string
+        try:
+            data_pickled = codecs.decode(self.proc_data_bytes, "base64")
+        except binascii.Error:
+            raise ValueError("Incorrect padding for data length %d bytes" %
+                             len(self.proc_data_bytes), "(mod 4 = %d)" %
+                             len(self.proc_data_bytes) % 4)
```

Comment:
it feels to me that you an have simple unit tests for stuff like this. But I wouldn't push it if you don't agree